bin/bl-exit

```diff
@@ -1,154 +1,425 @@
-        self.set_icon(windowicon)
-
-        # Cancel key (Escape)
-        accelgroup = gtk.AccelGroup()
-        key, mod = gtk.accelerator_parse('Escape')
-        accelgroup.connect_group(key, mod, gtk.ACCEL_VISIBLE, gtk.main_quit)
-        self.add_accel_group(accelgroup)
-
-        self._construct_ui()
-
-    def _construct_ui(self):
-        self._button_box = gtk.HBox()
-        self._add_button(stock=gtk.STOCK_CANCEL, action=self._cancel_action)
-        self._add_button(label="_Log out", action=self._logout_action)
-        self._add_button(label="_Suspend", action=self._suspend_action)
-        self._add_button(label="Re_boot", action=self._reboot_action)
-        self._add_button(label="_Power off", action=self._shutdown_action)
-
-        self._status = gtk.Label()
-        label_box = gtk.HBox()
-        label_box.pack_start(self._status)
-
-        vbox = gtk.VBox()
-        vbox.pack_start(self._button_box)
-        vbox.pack_start(label_box)
-
-        self.add(vbox)
-        self.show_all()
+    def do_action(self, action):
+        print("do_action: {}".format(action), file=sys.stderr)
+        self.send_dbus(action)
+
+    def send_dbus(self, method):
+        try:
+            if self.dbus_iface is None:
+                self.setup_dbus_connection()
+            if method[:3] == "Can":
+                command = "self.dbus_iface.{}()".format(method)
+            else:
+                command = "self.dbus_iface.{}(['True'])".format(method)
+            response = eval(command)
+            return str(response)
+        except dbus.DBusException as e:
+            self.on_error(str(e))
+
+    def on_error(self, string):
+        print ("{}".format(string), file=sys.stderr)
+        sys.exit(1)
+
+    def openbox_exit(self):
+        subprocess.check_output(["openbox", "--exit"])
+
+    def logout(self):
+        try:
+            self.openbox_exit()
+        except subprocess.CalledProcessError as e:
+            self.on_error(e.output)
+
+    def action_from_command_line(self, action):
+        try:
+            self.do_action(action)
+        except (subprocess.CalledProcessError, CanDoItError, KeyError) as e:
+            self.on_error(str(e))
+
+    def print_usage(self, status):
+        print ("bl-exit version {} usage:".format(__version__),
+               "  -h, --help        show this message and exit", sep='\n',
+               file=sys.stderr)
+        if display:
+            print ("  -l, --logout      log out from openbox", sep='\n',
+                   file=sys.stderr)
+        print ("  -s, --suspend     suspend the system",
+               "  -i, --hibernate   hibernate the system",
+               "  -y, --hybridsleep hibernate + sleep the system",
+               "  -b, --reboot      reboot the system",
+               "  -p, --poweroff    power the system down", sep='\n', end='\n',
+               file=sys.stderr)
+        sys.exit(status)
 
     def main(self):
```

Comment:
You can get rid of all the CLI handling by using ArgumentParser from the argparse module which is part of the stdlib. It will do everything for you, including the generation of a --help message.